http_handlers.go

```diff
@@ -102,6 +140,106 @@ func getMetadata(filter string) http.HandlerFunc {
 	}
 }
 
+func ec2Handler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		return
+	}
+
+	logger.Debug("Calling ec2Handler ")
+	userIP := getIPFromRequest(r)
+	if userIP != "" {
```

Comment:
return early if `userIP == ""` and avoid a level of indentation

Reply:
Oops, forgot to change this. Will do now.